core/domain/learner_goals_services.py

```diff
@@ -0,0 +1,143 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for the learner goals feature of the learner dashboard."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.domain import user_domain
+from core.platform import models
+import feconf
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+MAX_CURRENT_GOALS_COUNT = feconf.MAX_CURRENT_GOALS_COUNT
+
+
+def get_learner_goals_from_model(learner_goals_model):
+    """Returns the learner goals domain object given the learner goals
+    model loaded from the datastore.
+
+    Args:
+        learner_goals_model: LearnerGoalsModel. The
+            learner goals model from the datastore.
+
+    Returns:
+        LearnerGoals. The learner goals domain object corresponding to the
+        given model.
+    """
+    return user_domain.LearnerGoals(
+        learner_goals_model.id,
+        learner_goals_model.topic_ids_to_learn,
+        learner_goals_model.topic_ids_to_master)
+
+
+def save_learner_goals(learner_goals):
+    """Save a learner goals domain object as an LearnerGoalsModel entity
+    in the datastore.
+
+    Args:
+        learner_goals: LearnerGoals. The learner goals domain object to
+            be saved in the datastore.
+    """
+    learner_goals_dict = {
+        'topic_ids_to_learn': learner_goals.topic_ids_to_learn
+    }
+
+    learner_goals_model = (user_models.LearnerGoalsModel.get_by_id(
```

Comment:
Don't use `get_by_id`. I think that returns deleted models as well. Use `.get(id, strict=False)`. (You can see its definition in base_models)

Reply:
Done.